### PR TITLE
Added valid types to server_onupdate

### LINKED ISSUE #11546: server_onupdate don't allow valid types
### Ensure stubs packages are not installed

- [X] No sqlalchemy stub packages is installed (both `sqlalchemy-stubs` and `sqlalchemy2-stubs` are not compatible with v2)

### Verify if the api is typed

- [X] The api is not in a module listed in [#6810](https://github.com/sqlalchemy/sqlalchemy/issues/6810) so it should pass type checking

### Describe the typing issue

When defining `server_onupdate=<...>` in `Column()` or in `mapped_column()`, valid types are `FetchedValue`, `str`, `ClauseElement`, `TextClause`.

However, the kwarg is typed as `Optional[FetchedValue]` in both `Column()` and `mapped_column()`.

### To Reproduce

```python
from sqlalchemy import Column, DateTime, Table, func, text
from sqlalchemy.orm import DeclarativeBase

class Base(DeclarativeBase):
    pass

my_table = Table(
    "my_table",
    Base.metadata,
    Column("updated_at1", DateTime, server_onupdate=func.now()),
    Column("updated_at2", DateTime, server_onupdate=text("NOW")),
    Column("updated_at3", DateTime, server_onupdate="NOW"),
)
```

### Error

```
main.py:12: error: Argument "server_onupdate" to "Column" has incompatible type "now"; expected "FetchedValue | None"  [arg-type]
main.py:13: error: Argument "server_onupdate" to "Column" has incompatible type "TextClause"; expected "FetchedValue | None"  [arg-type]
main.py:14: error: Argument "server_onupdate" to "Column" has incompatible type "str"; expected "FetchedValue | None"  [arg-type]
Found 3 errors in 1 file (checked 1 source file)
``` 

### Versions

- OS: Arch Linux
- Python: 3.12.3
- SQLAlchemy: 2.0.31
- Type checker: mypy 1.10.1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by opkna**
Fixes: #11546

### Description
* Added `str`, `TextClause`, `ColumnElement[Any]` as valid types for `server_onupdate` argument in `Column()` and `mapped_column()`. Before it was only `FetchedValue`.
* Expanded mypy tests for `Column()`
* Created new mypy test file specifically for testing `mapped_column` kwargs

### Checklist

This pull request is:

- [ ] A documentation / typographical / small typing error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: lib/sqlalchemy/sql/schema.py [hunk 1]
```diff
@@ -126,6 +126,10 @@
     "FetchedValue", str, TextClause, ColumnElement[Any]
 ]
 
+_ServerOnUpdateArgument = Union[
```
Comment (CaselIT) [inline lib/sqlalchemy/sql/schema.py:129]: Maybe we can re-use the `_ServerDefaultArgument` type (could be renamed)

@zzzeek do you prefer having two aliases here or just one?

File: test/typing/plain_files/orm/mapped_column_kwargs.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+from sqlalchemy import Boolean
```
Comment (CaselIT) [inline test/typing/plain_files/orm/mapped_column_kwargs.py:1]: this seems very similar to https://github.com/sqlalchemy/sqlalchemy/blob/main/test/typing/plain_files/orm/mapped_column.py as scope, so maybe you can move the same come there?

**Review by CaselIT — COMMENTED** [review]
great work, thanks! Left a couple of comments

File: lib/sqlalchemy/sql/schema.py (hunk 1 above)
Comment (opkna) [reply lib/sqlalchemy/sql/schema.py:129]: Only hard thing about that is finding a good name for it

File: test/typing/plain_files/orm/mapped_column_kwargs.py (hunk 2 above)
Comment (opkna) [reply test/typing/plain_files/orm/mapped_column_kwargs.py:1]: Sure I can merge them

File: lib/sqlalchemy/sql/schema.py (hunk 1 above)
Comment (CaselIT) [reply lib/sqlalchemy/sql/schema.py:129]: that works too

Comment (opkna) [reply lib/sqlalchemy/sql/schema.py]: Updated!

**Comment by CaselIT** [comment]
thanks!

**Pull request merged**